pkg/users/account-details.js

```diff
@@ -343,3 +334,104 @@
         </Page>
     );
 }
+
+export const AccountGroupsSelect = ({ name, groups, setError }) => {
+    const usedGroups = groups.filter(group => group.userlist.includes(name));
+    const primaryGroup = groups.find(group => group.userlistPrimary.includes(name));
+    const primaryGroupName = primaryGroup ? primaryGroup.name : undefined;
+    const [isOpenGroup, setIsOpenGroup] = useState(false);
+    const _selected = usedGroups.map(group => group.name);
+    if (primaryGroup)
+        _selected.push(primaryGroupName);
+    const [selected, setSelected] = useState(_selected);
+    const [validated, setValidated] = useState('default');
+    const previousValue = useRef(null);
+
+    useEffect(() => {
+        const items_diff = previousValue.current
+            ? selected
+                    .filter(x => !previousValue.current.includes(x))
+                    .concat(previousValue.current.filter(x => !selected.includes(x)))
+            : [];
+        const validated = items_diff.length ? 'warning' : 'default';
+        setValidated(validated);
+        previousValue.current = selected;
+    }, [selected, previousValue, setValidated]);
+
+    const removeGroup = group => {
+        return cockpit.spawn(["/usr/sbin/usermod", "-r", "-G", group, name], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected(selected.filter(item => item !== group));
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const addGroup = group => {
+        return cockpit.spawn(["/usr/sbin/usermod", "-a", "-G", group, name], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected([...selected, group]);
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const onSelectGroup = (event, selection) => {
+        if (selected.includes(selection)) {
+            removeGroup(selection);
+        } else {
+            addGroup(selection);
+        }
+    };
+
+    const chipGroupComponent = () => {
+        return (
+            <LabelGroup numLabels={10}>
+                {(selected || []).map((currentLabel, index) => {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17996-20221206-122041-ea567d53-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)